Provider crashes when doing a data source lookup of an app with different type than the label it is using for the lookup.

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform Version

<!--- Please run `terraform -v` to show the Terraform core version and provider version(s). If you are not running the latest version of Terraform or the provider, please upgrade because your issue may have already been fixed. [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions). --->

Terraform version: `1.1.9`

Provider version: `3.24.0`

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

- `okta_app_oidc` data source
- `okta_app_bookmark` data source 

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

```hcl

resource "okta_app_oauth" "foo" {
  label                     = "foo OIDC Backend [${local.eks_cluster_name}]"
  hide_ios                  = true
  hide_web                  = true
  type                      = "native"
  omit_secret               = false
  grant_types               = ["authorization_code"]
  response_types            = ["code"]
  redirect_uris             = ***
  post_logout_redirect_uris = ***
  consent_method            = "TRUSTED"
  # Enable PKCE support
  token_endpoint_auth_method = "none"

  # Inject groups claim in tokens
  groups_claim {
    type        = "FILTER"
    filter_type = "STARTS_WITH"
    name        = "groups"
    value       = ***
  }

  lifecycle {
    ignore_changes = [
      groups
    ]
  }
}

resource "okta_app_bookmark" "foo {
  label = "foo [${local.eks_cluster_name}]"
  url   = ***
  logo  = ***

  lifecycle {
    ignore_changes = [
      groups
    ]
  }
}

data "okta_app_oauth" "foo" {
  label = "foo [${local.eks_cluster_name}]"
}
```

### Panic Output

<!--- If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`. --->

```
The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadDataSource call. The plugin logs may contain more details.

Stack trace from the terraform-provider-okta_v3.24.0 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x174a003]

goroutine 97 [running]:
github.com/okta/terraform-provider-okta/okta.dataSourceAppOauthRead({0x1b390a8, 0xc000246660}, 0x23043c8, {0x18d1860, 0xc0003cb860})
        github.com/okta/terraform-provider-okta/okta/data_source_okta_app_oauth.go:196 +0x7e3
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc0004d8460, {0x1b390e0, 0xc0008273b0}, 0xd, {0x18d1860, 0xc0003cb860})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.13.0/helper/schema/resource.go:725 +0x12e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).ReadDataApply(0xc0004d8460, {0x1b390e0, 0xc0008273b0}, 0xc0002bf080, {0x18d1860, 0xc0003cb860})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.13.0/helper/schema/resource.go:944 +0x145
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadDataSource(0xc0000cc180, {0x1b39038, 0xc00021c280}, 0xc0000e4620)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.13.0/helper/schema/grpc_provider.go:1179 +0x408
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadDataSource(0xc0000d00a0, {0x1b390e0, 0xc000826480}, 0xc00060c0f0)
        github.com/hashicorp/terraform-plugin-go@v0.8.0/tfprotov5/tf5server/server.go:657 +0x45f
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadDataSource_Handler({0x19ad400, 0xc0000d00a0}, {0x1b390e0, 0xc000826480}, 0xc000246120, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.8.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:421 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0002a8a80, {0x1b48e90, 0xc000001a00}, 0xc000630480, 0xc0000bcba0, 0x20f8a30, 0x0)
        google.golang.org/grpc@v1.45.0/server.go:1282 +0xccf
google.golang.org/grpc.(*Server).handleStream(0xc0002a8a80, {0x1b48e90, 0xc000001a00}, 0xc000630480, 0x0)
        google.golang.org/grpc@v1.45.0/server.go:1619 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.45.0/server.go:921 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.45.0/server.go:919 +0x294

Error: The terraform-provider-okta_v3.24.0 plugin crashed!
```

### Expected Behavior

<!--- What should have happened? --->

The provider should not crash when it looks up a conflicting data source and should instead just post an message indicating the config error.

### Actual Behavior

<!--- What actually happened? --->

The plugin crashes.

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. Create two Okta apps. In my case, an `okta_app_oauth` and `okta_app_bookmark`
2. Give the apps different labels to identify their purpose like `foo OIDC Backend ` and `foo` respectively.
3. Do a data source lookup on one of the apps, but mismatch the label with the app type
   ```
   data "okta_app_oidc" "foo" {
     label = "foo"
   }
   ```
4. Run `terraform plan` and observe the crash.

--- a/okta/data_source_okta_app_oauth.go
+++ b/okta/data_source_okta_app_oauth.go
@@ -193,34 +193,45 @@ func dataSourceAppOauthRead(ctx context.Context, d *schema.ResourceData, m inter
 	_ = d.Set("label", app.Label)
 	_ = d.Set("name", app.Name)
 	_ = d.Set("status", app.Status)
-	_ = d.Set("type", app.Settings.OauthClient.ApplicationType)
 	_ = d.Set("auto_submit_toolbar", app.Visibility.AutoSubmitToolbar)
 	_ = d.Set("hide_ios", app.Visibility.Hide.IOS)
 	_ = d.Set("hide_web", app.Visibility.Hide.Web)
-	_ = d.Set("client_uri", app.Settings.OauthClient.ClientUri)
-	_ = d.Set("logo_uri", app.Settings.OauthClient.LogoUri)
-	_ = d.Set("login_uri", app.Settings.OauthClient.InitiateLoginUri)
-	_ = d.Set("client_id", app.Credentials.OauthClient.ClientId)
-	_ = d.Set("policy_uri", app.Settings.OauthClient.PolicyUri)
-	_ = d.Set("wildcard_redirect", app.Settings.OauthClient.WildcardRedirect)
-	respTypes := make([]string, len(app.Settings.OauthClient.ResponseTypes))
-	for i := range app.Settings.OauthClient.ResponseTypes {
-		respTypes[i] = string(*app.Settings.OauthClient.ResponseTypes[i])
+
+	respTypes := []string{}
+	grantTypes := []string{}
+	redirectUris := []string{}
+	postLogoutRedirectUris := []string{}
+
+	if app.Settings.OauthClient != nil {
+		_ = d.Set("type", app.Settings.OauthClient.ApplicationType)
+		_ = d.Set("client_uri", app.Settings.OauthClient.ClientUri)
+		_ = d.Set("logo_uri", app.Settings.OauthClient.LogoUri)
+		_ = d.Set("login_uri", app.Settings.OauthClient.InitiateLoginUri)
+		_ = d.Set("client_id", app.Credentials.OauthClient.ClientId)
+		_ = d.Set("policy_uri", app.Settings.OauthClient.PolicyUri)
+		_ = d.Set("wildcard_redirect", app.Settings.OauthClient.WildcardRedirect)
+		for i := range app.Settings.OauthClient.ResponseTypes {
+			respTypes = append(respTypes, string(*app.Settings.OauthClient.ResponseTypes[i]))
+		}
+		for i := range app.Settings.OauthClient.GrantTypes {
+			grantTypes = append(grantTypes, string(*app.Settings.OauthClient.GrantTypes[i]))
+		}
+		redirectUris = append(redirectUris, app.Settings.OauthClient.RedirectUris...)
+		postLogoutRedirectUris = append(postLogoutRedirectUris, app.Settings.OauthClient.PostLogoutRedirectUris...)
 	}
-	grantTypes := make([]string, len(app.Settings.OauthClient.GrantTypes))
-	for i := range app.Settings.OauthClient.GrantTypes {
-		grantTypes[i] = string(*app.Settings.OauthClient.GrantTypes[i])
-	}
+
 	aggMap := map[string]interface{}{
-		"redirect_uris":             convertStringSliceToSet(app.Settings.OauthClient.RedirectUris),
+		"redirect_uris":             convertStringSliceToSet(redirectUris),
 		"response_types":            convertStringSliceToSet(respTypes),
 		"grant_types":               convertStringSliceToSet(grantTypes),
-		"post_logout_redirect_uris": convertStringSliceToSet(app.Settings.OauthClient.PostLogoutRedirectUris),
+		"post_logout_redirect_uris": convertStringSliceToSet(postLogoutRedirectUris),
 	}
-	if app.Settings.OauthClient.IdpInitiatedLogin != nil {
+	if app.Settings.OauthClient != nil &&
+		app.Settings.OauthClient.IdpInitiatedLogin != nil {
 		_ = d.Set("login_mode", app.Settings.OauthClient.IdpInitiatedLogin.Mode)
 		aggMap["login_scopes"] = convertStringSliceToSet(app.Settings.OauthClient.IdpInitiatedLogin.DefaultScope)
 	}
+
 	err = setNonPrimitives(d, aggMap)
 	if err != nil {
 		return diag.Errorf("failed to set OAuth application properties: %v", err)

--- a/website/docs/d/app_oauth.html.markdown
+++ b/website/docs/d/app_oauth.html.markdown
@@ -20,9 +20,13 @@ data "okta_app_oauth" "test" {
 
 ## Argument Reference
 
-- `label` - (Optional) The label of the app to retrieve, conflicts with `label_prefix` and `id`. Label uses
-  the `?q=<label>` query parameter exposed by Okta's API. It should be noted that at this time this searches both `name`
-  and `label`. This is used to avoid paginating through all applications.
+- `label` - (Optional) The label of the app to retrieve, conflicts with
+    `label_prefix` and `id`. Label uses the `?q=<label>` query parameter exposed by
+    Okta's List Apps API. The API will search both `name` and `label` using that
+    query. Therefore similarily named and labeled apps may be returned in the query
+    and have the unitended result of associating the wrong app with this data
+    source. See:
+    https://developer.okta.com/docs/reference/api/apps/#list-applications
 
 - `label_prefix` - (Optional) Label prefix of the app to retrieve, conflicts with `label` and `id`. This will tell the
   provider to do a `starts with` query as opposed to an `equals` query.

